### PR TITLE
Create competitions

### PR DESCRIPTION
**Pull request by mthembbu**

### CODE REVIEW

File: skietbaan_fe/src/components/CreateComp.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		this.setState({ [e.target.name]: e.target.value });
+	}
+	//------------------------------------------------------------------------
+	/** A method that handles the submit enent for the submit button*/
+	onSubmit(e) {
+		/** Preventing the default button action event to occur automatically*/
+		e.preventDefault();
+		const compData = {
+			compTitle: this.state.compTitle,
+			compStatus: true
```
Comment (laurent-mbuyu) [inline skietbaan_fe/src/components/CreateComp.js]: compStatus seems to be a boolean variable.
Set its initial value to false if you intend to toggle it

File: skietbaan_fe/src/components/CreateComp.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	constructor() {
+		super();
+		this.state = { compTitle: '', compStatus: '' };
+		//binding the onChange method to this commponents
+		this.onChange = this.onChange.bind(this);
+		this.onSubmit = this.onSubmit.bind(this);
+	}
+	//------------------------------------------------------------------------
+	/** A method that detects the change in the change in thw textfield */
+	onChange(e) {
```
Comment (laurent-mbuyu) [inline skietbaan_fe/src/components/CreateComp.js]: e is easy to easy and faster but make it more meaningul.
keep it simple by replacing it with event in your upcoming event manipulation

File: skietbaan_fe/src/components/CreateComp.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+		this.setState({ [e.target.name]: e.target.value });
+	}
+	//------------------------------------------------------------------------
+	/** A method that handles the submit enent for the submit button*/
+	onSubmit(e) {
+		/** Preventing the default button action event to occur automatically*/
+		e.preventDefault();
+		const compData = {
```
Comment (laurent-mbuyu) [inline skietbaan_fe/src/components/CreateComp.js]: Why is this variable a const?
Won't the title change later on?

File: skietbaan_fe/src/components/CreateComp.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+			compTitle: this.state.compTitle,
```
Comment (laurent-mbuyu) [inline skietbaan_fe/src/components/CreateComp.js]: Are you setting the state of this.state.compTitle beside its initial state?

File: skietbaan_fe/src/components/CreateComp.js (hunk 1 above)
Comment (mthembbu) [reply skietbaan_fe/src/components/CreateComp.js]: (True) is a default value

File: skietbaan_fe/src/components/CreateComp.js (hunk 4 above)
Comment (mthembbu) [reply skietbaan_fe/src/components/CreateComp.js]: I am trying to assign the compTitle to the compTitle using a react syntax, or am I doing it wrong

File: skietbaan_fe/src/components/CreateComp.js (hunk 3 above)
Comment (mthembbu) [reply skietbaan_fe/src/components/CreateComp.js]: That compData, is an initialised competition object that will be transferred to createComp(compData)

File: skietbaan_fe/src/components/CreateComp.js (hunk 1 above)
Comment (laurent-mbuyu) [reply skietbaan_fe/src/components/CreateComp.js]: It has the same name as one of the state properties. I see.

File: skietbaan_fe/src/components/CreateComp.js (hunk 2 above)
Comment (mthembbu) [reply skietbaan_fe/src/components/CreateComp.js]: I here you, but the onChange method caters for other Text inputs that may be included in the future

File: skietbaan_fe/src/components/CreateComp.js (hunk 4 above)
Comment (laurent-mbuyu) [reply skietbaan_fe/src/components/CreateComp.js]: By react syntax you mean using the value of the text field in the form. I see.

File: skietbaan_fe/src/components/CreateComp.js (hunk 2 above)
Comment (laurent-mbuyu) [reply skietbaan_fe/src/components/CreateComp.js]: Cool

File: skietbaan_fe/src/components/CreateComp.js (hunk 3 above)
Comment (laurent-mbuyu) [reply skietbaan_fe/src/components/CreateComp.js]: Great

**Pull request merged**